crates/apollo-router-core/src/query_planner/model.rs

```diff
@@ -4,18 +4,25 @@
 //! QueryPlans are a set of operations that describe how a federated query is processed.
 
 use crate::prelude::graphql::*;
-use serde::{Deserialize, Serialize};
+use serde::Deserialize;
 
 /// The root query plan container.
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Debug, PartialEq, Deserialize)]
 #[serde(tag = "kind")]
 pub struct QueryPlan {
     /// The hierarchical nodes that make up the query plan
-    pub node: Option<PlanNode>,
+    node: Option<PlanNode>,
+}
+
+impl QueryPlan {
+    /// Returns a reference to the plan.
+    pub fn node(&self) -> Option<&PlanNode> {
+        self.node.as_ref()
+    }
```

Comment:
I'm not a big fan of getters but in this case this field is really not meant to be mutated.

Reply:
using a getter here is a good idea, we should not get into the habit of messing with the query plan's internals in the fetching phase, since the query plan will likely change a lot in the future